vsutil/clips.py

```diff
@@ -113,8 +113,22 @@ def insert_clip(clip: vs.VideoNode, /, insert: vs.VideoNode, start_frame: int) -
 
 def join(planes: List[vs.VideoNode], family: vs.ColorFamily = vs.YUV) -> vs.VideoNode:
     """
-    Joins the supplied list of planes into a three-plane VideoNode (defaults to YUV).
-    """
+    Joins the supplied list of planes into a VideoNode containing the given planes.
+    
+    :param family: The color-family the resulting VideoNode will advertise.
+    :return: A VideoNode whose individual planes match the planes passed to this function.
+    """
+    if len(planes) == 1:
+        # Equivalent to planes[0] but is supported by more types.
+        plane_clip = next(iter(planes))
+        if family != vs.GRAY:
+            raise ValueError('If passing only a single plane to join, the output family must be vs.GRAY')
+
+        # This should probably be an error to use join like this, but we let it pass.
+        if plane_clip.format.num_planes != 1:
+            plane_clip = plane(plane_clip, 0)
+        
+        return plane_clip
```

Comment:
This seems really over-engineered, when really all that has to be done is  making a single-item join a no-op by returning *planes[0]*

The function could be simply:

```py
def join(planes: List[vs.VideoNode], family: vs.ColorFamily = vs.YUV) -> vs.VideoNode:
    for plane in planes:
        if plane.format.num_planes > 1:
            raise ValueError("join: all planes should be single-plane grayscale VideoNodes")
    if len(planes) not in [1, 3]:
      raise ValueError("join: planes argument must be exactly 1 or 3 items long")
      
    return planes[0] if len(planes) == 1 else core.std.ShufflePlanes(planes, planes=[0, 0, 0], colorfamily=family)
```

and arguably, we don't really need to check if the clips are single-plane, so it could be as simple as

```py
def join(planes: List[vs.VideoNode], family: vs.ColorFamily = vs.YUV) -> vs.VideoNode:
    if len(planes) not in [1, 3]:
      raise ValueError("join: planes argument must be exactly 1 or 3 items long")

    return planes[0] if len(planes) == 1 else core.std.ShufflePlanes(planes, planes=[0, 0, 0], colorfamily=family)
```

The check for number of planes given though has to stay, since `core.std.ShufflePlanes([Y,U], planes=[0,0,0], colorfamily=vs.YUV)` is a valid (but not desirable) call.